DIPs/DIP-31.md

```diff
@@ -76,4 +76,132 @@ The PCD Team of 0xPARC is responsible for developing, testing, and deploying thi
+
+## PCDpass Pretix Two-Way Sync
+
+In order to be able to issue PCD-based tickets to event attendees, PCDpass needs to be aware of the `Organizer`, `Event`s, and `Product`s that live in Pretix. Thus, we have developed a syncing mechanism that downloads state from Pretix and saves it to our own database in our own data format.
+
+Internally, we refer to an individual ticket to an event as a `Pretix Ticket`. A more explicit definition of `Pretix Ticket` can be found in the description of the algorithm by which we sync an `Organizer` into PCDpass.
+
+- A ticket can either be `Soft Deleted` or not.
+
+The sync algorithm is as follows:
+
+- For each `Organizer` that has been configured, and for each `Event` that has been specified as relevant, we download the `Event` metadata.
+    - For each `Event`, we download its name, which I will refer to as its `event_name`.
+    - For each `Product` within each `Event` that PCDpass understands to be a ticket `Product`, as identified by that `Event`'s `active_item_ids`, we sync that `Product`'s name, which I will refer to as its `item_name`.
+- For each `Event`, we download all the `Order`s.
+- For each `Order`, we extract the `Position`s.
+- We record each `Position`  as a `Pretix Ticket` in our system.
+- The sync algorithm is robust to `Order`s and `Position`s changing.
+    - In the case that an `Order` containing the `Position` that caused us to record a `Pretix Ticket` is cancelled or deleted, we record the `Pretix Ticket` as `Soft Deleted`.
+    - In the case that the `Position` that caused us to record a `Pretix Ticket` is cancelled or deleted, we record the `Pretix Ticket` as `Soft Deleted`.
+    - In the case that we encounter one or more `Position`s that translated to a `Pretix Ticket` that is `Soft Deleted`, we make that `Pretix Ticket` **not** `Soft Deleted`.
+- The sync algorithm is robust to to `Event`s and `Product`s being renamed.
+- The sync algorithm is robust to `Event`s and `Product`s being deleted.
+    - In the case that an `Event` or `Product` no longer exists, we cancel the sync so that existing `Pretix Ticket`s are not changed. Fixing this state requires manual intervention by the PCD Team.
+- For each `Pretix Ticket` that has been checked in by an event volunteer on-site, we sync the check-in to the corresponding `Position` in Pretix. In the case that the ticket was checked in both on the Pretix side and on the PCDpass side, and had its checkin deleted by an organizer on Pretix, the ticket will become un-checked-in on PCDpass as well, letting the attendee get checked in by an event volunteer via PCDpass. To summarize, each ticket's check-in status is synced between PCDpass and Pretix, such that the latest state is in sync between the two systems of record.
```

Comment:
maybe just do the summary on the end and delete the beginning. The summary sounds sane to me - the stuff before not so much - e.g.
> In the case that the ticket was checked in both on the Pretix side and on the PCDpass side
IMHO this should never be able - you should only be able to check in with one method - when then trying with the other it should fail as the ticket is already checked in

Reply:
I think we are in agreement - 

- checking in a ticket that has not been checked in on either Pretix or PCDpass will work.
- checking in a ticket that has been checked on the other platform will not work. I'll update this description to be more clear.